ambry-utils/src/test/java/com.github.ambry.utils/SimpleByteBufferPoolTest.java

```diff
@@ -74,45 +80,51 @@ public void testDeallocateExceedPoolCapacity()
   }
 
   /**
-   * Scenario: timeout after not enough memory.
+   * This scenario tests when a {@link SimpleByteBufferPool} tries to allocate a {@link ByteBuffer} that is smaller
+   * than the pool's capacity, but larger than the pool's current available memory. The operation will be blocking
+   * until eventually be timed out and throw {@link TimeoutException}.
    */
   @Test
   public void testNotEnoughMemory()
-      throws InterruptedException, TimeoutException {
+      throws Exception {
     final int size = 1024;
     final long capacity = 1024;
-    final long maxBlockTimeInMs = 20;
+    final long maxBlockTimeInMs = 10;
     SimpleByteBufferPool pool = new SimpleByteBufferPool(capacity);
-    ByteBuffer buffer1 = pool.allocate(size, maxBlockTimeInMs);
+    pool.allocate(size, maxBlockTimeInMs);
     try {
-      ByteBuffer buffer2 = pool.allocate(size, maxBlockTimeInMs);
+      pool.allocate(size, maxBlockTimeInMs);
       fail("should have thrown.");
     } catch (TimeoutException e) {
     }
   }
 
   /**
-   * Scenario: zero and negative timeout.
+   * This test ensures that the {@link SimpleByteBufferPool} behaves correctly when the {@code timeToBlockInMs}
+   * parameter set in {@link SimpleByteBufferPool#allocate(int, long)} is zero or a negative value.
    */
   @Test
   public void testNegativeBlockTime()
-      throws InterruptedException, TimeoutException {
+      throws Exception {
     final int size = 1024;
     final long capacity = 1024;
     final long maxBlockTimeInMs = 20;
     SimpleByteBufferPool pool = new SimpleByteBufferPool(capacity);
-    ByteBuffer buffer=pool.allocate(size, 0);
+    ByteBuffer buffer = pool.allocate(size, 0);
     pool.deallocate(buffer);
-    try{
-      buffer=pool.allocate(size, -maxBlockTimeInMs);
-    }catch(IllegalArgumentException e){
+    try {
+      buffer = pool.allocate(size, -maxBlockTimeInMs);
+    } catch (IllegalArgumentException e) {
```

Comment:
Yes, this `IllegalArgumentException` is expected when size is greater than buffer pool's capacity or `maxBlockTimeInMs` is negative.